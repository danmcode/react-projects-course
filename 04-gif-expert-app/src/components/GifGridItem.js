import React from 'react';
//impt
import PropTypes from 'prop-types'; //Manejo de properties


export const GifGridItem = ( { id, title, url } ) => {
  return (
    <div className='card animate__animated animate__fadeIn'>
        <img src={ url } alt={ title }/>
        <p> { title } </p>
    </div>
  );
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}


/**
 * Configuracion de tests
 * 1. Enzyme
 * 2. Enzyme to Json
 * 3. Debe mostrar el componente correctamente
 * 4. Implementando protypes.
 *  * shallow
 *  * wrapper
 *  * wrapper .toMatchSnapshot
 */