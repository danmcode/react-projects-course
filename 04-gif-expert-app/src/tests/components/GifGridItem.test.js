import { shallow } from 'enzyme';
import React from 'react'
import { GifGridItem } from '../../components/GifGridItem'

describe('test to <GifGridItem />', () => {

    const title = "Un titulo";
    const url = "https://localhost/unaimagen.jpg";

    beforeEach(()=>{
        wrapper = shallow(        
        <GifGridItem 
            title={ title }
            url = { url }
        /> );
    });
    
    let wrapper = shallow( 
        <GifGridItem 
            title={ title }
            url = { url }
        /> 
    );

    test('should render correct the component', () => { 
        expect( wrapper ).toMatchSnapshot();
     });
 });