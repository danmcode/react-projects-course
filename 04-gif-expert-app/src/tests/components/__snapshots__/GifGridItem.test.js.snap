// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to <GifGridItem /> should render correct the component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/unaimagen.jpg"
  />
  <p>
     
    Un titulo
     
  </p>
</div>
`;
